src/main/handlers/media/transcriptionEngines/voskTranscribeFunction.ts

```diff
@@ -0,0 +1,28 @@
+import { MapCallback, PartialWord } from 'sharedTypes';
+import fs from 'fs';
+import path from 'path';
+import { TranscriptionFunction } from '../transcribeTypes';
+
+interface VoskWord {
+  end: number;
+  start: number;
+  word: string;
+}
+
+const voskAdaptor: MapCallback<VoskWord, PartialWord> = (result) => ({
```

Comment:
Is this adapting the vosk output to the 'generic' transcript you mentioned? And then after this it is passed to preProcess?

Reply:
Yea in this PR i tried to remove the adaptor stuff in preProcess and move it into the transcription functions, hence why the cammelCase function is now in the deepspeech and dummy engine files. I think most of the stuff left in preProcess is now actual preprocessing, not adaptor stuff, but I could be wrong on that

Reply:
Ahh I see, that seems a lot cleaner. I suppose if you didn't do the adapting over here you'd end up having a messy preProcess which has to deal with various input shapes.
Happy for this to be merged!